plugins/ScheduledReports/API.php

```
@@ -381,6 +381,7 @@ public function generateReport(
         $reports = $this->getReports($idSite = false, $_period = false, $idReport);
         $report = reset($reports);
 
+        $idReport = $report['idreport'];
```

Should we maybe add a check here and throw an exception if a report was not found? Otherwise I guess the code below might causes notices or warnings when trying to access the array properties.

---

Add a small test for the case when `getReports` does not find anything useful. It was already throwing, but having an API test to ensure those errors are passed should not hurt.